flink-core/src/main/java/org/apache/flink/core/plugin/PluginManagerSingleton.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.core.plugin;
+
+import org.apache.flink.util.Preconditions;
+
+import java.util.Collection;
+
+/**
+ * Singleton instance provider of {@link PluginManager}. This class must be initialized before usage, ideally at the
+ * entry-point of each process that wants to use it.
+ */
+public final class PluginManagerSingleton {
```

Comment:
nit: maybe it's worth merging content of this class with `PluginManager`? If you prefer to keep them separated maybe java doc link the `PluginManagerSingleton` somewhere in the `PluginManager`?